xyz

```diff
@@ -117,62 +122,69 @@ inc() {
   printf %s "$version"
 }
 
-branch=master
-edit=false
-increment=patch
-message_template='Version X.Y.Z'
-prerelease_label=rc
-publish_command='npm publish'
-repo=origin
+declare -A options=(
+  [branch]=master
+  [dry-run]=false
+  [edit]=false
+  [increment]=patch
+  [message]='Version X.Y.Z'
+  [prerelease-label]=rc
+  [publish-command]='npm publish'
+  [repo]=origin
+  [tag]=vX.Y.Z
+)
 declare -a scripts
-tag_template=vX.Y.Z
-dry_run=false
 
 while (( $# > 0 )) ; do
-  option="$1"
-  shift
+  option="$1" ; shift
+
+  case $option in
+    -b) option=--branch ;;
+    -e) option=--edit ;;
+    -h) option=--help ;;
+    -i) option=--increment ;;
+    -m) option=--message ;;
+    -r) option=--repo ;;
+    -t) option=--tag ;;
+    -v) option=--version ;;
+  esac
 
-  case "$option" in
-    -h|--help)
-      echo "$usage"
-      exit
-      ;;
-    -v|--version)
-      node -p 'require("xyz/package.json").version'
-      exit
-      ;;
-    -b|--branch)            branch="$1"           ; shift ;;
-    -e|--edit)              edit=true             ;;
-    -i|--increment)         increment="$1"        ; shift ;;
-    -m|--message)           message_template="$1" ; shift ;;
-       --prerelease-label)  prerelease_label="$1" ; shift ;;
-       --publish-command)   publish_command="$1"  ; shift ;;
-    -r|--repo)              repo="$1"             ; shift ;;
-    -s|--script)            scripts+=("$1")       ; shift ;;
-    -t|--tag)               tag_template="$1"     ; shift ;;
-       --dry-run)           dry_run=true          ;;
+  case $option in
+    --help)
+      echo "$usage" ; exit ;;
+    --version)
+      node -p 'require("xyz/package.json").version' ; exit ;;
+    --script)
+      scripts+=("$1") ; shift ;;
+    --dry-run|--edit)
+      options[${option:2}]=true ;;
```

Comment:
I can't find the `${parameter:value}`-syntax in [this page](http://www.tldp.org/LDP/abs/html/parameter-substitution.html). Only  `parameter:[-=+?]value`. What does this do?

Reply:
It is substring extraction (i.e. string slicing). See [`${string:position}`][1] and [`${string:position:length}`][2]. In this case we're just stripping the leading hyphens.


[1]: http://www.tldp.org/LDP/abs/html/string-manipulation.html#SUBSTREXTR01
[2]: http://www.tldp.org/LDP/abs/html/string-manipulation.html#SUBSTREXTR02

Reply:
Ahhh! :)